File: leetcode/search/35. Search Insert Position/js/main.js
// Time O(LogN)
// Space O(1)
const searchInsert = (arr, search) => {
  let left = 0,
    right = arr.length - 1;

  while (left < right) {
    const mid = Math.floor((left + right) / 2);
    if (arr[mid] >= search) {
      right = mid;
    } else {
      left = mid + 1;
    }
  }
  return arr[right] < search ? right + 1 : right;
};

File: leetcode/search/35. Search Insert Position/py/main.py
# Time O(LogN)
# Space O(1)
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        lo, hi = 0, len(nums) - 1
        
        while lo < hi:
            mid = lo + ((hi - lo) // 2)
            
            if nums[mid] < target:
                lo = mid + 1
            else:
                hi = mid
        
        if nums[lo] < target:
            return lo + 1
        return lo